tornado/log.py

```diff
@@ -190,10 +190,27 @@ def enable_pretty_logging(options=None, logger=None):
         logger = logging.getLogger()
     logger.setLevel(getattr(logging, options.logging.upper()))
     if options.log_file_prefix:
-        channel = logging.handlers.RotatingFileHandler(
-            filename=options.log_file_prefix,
-            maxBytes=options.log_file_max_size,
-            backupCount=options.log_file_num_backups)
+        rotate_mode = options.log_rotate_mode
+        if rotate_mode == 'size':
+            channel = logging.handlers.RotatingFileHandler(
+                filename=options.log_file_prefix,
+                maxBytes=options.log_file_max_size,
+                backupCount=options.log_file_num_backups)
+        elif rotate_mode == 'time':
+            channel = logging.handlers.TimedRotatingFileHandler(
+                filename=options.log_file_prefix,
+                when=options.log_rotate_when,
+                interval=options.log_rotate_interval,
+                backupCount=options.log_file_num_backups)
+        else:
+            if not isinstance(rotate_mode, str):
+                error_message = 'The type of log_rotate_mode option should be ' +\
+                                'str, not {}.'.format(str(type(rotate_mode)))
```

Comment:
`{}` needs to be `{0}` for python 2.6 compatibility. (or use `%`-formatting, which is what we normally do in Tornado).

You don't need to check the type here; the options framework already guarantees that it's a string. Remove the TypeError and leave the ValueError.